tests/index.test.ts

```diff
@@ -0,0 +1,89 @@
+import { ExtensibilitySdk, Message, OAuthDialogCompletedMessage } from '../src/index';
+import runtime from '../src/sdk/RuntimeContext';
+import oauthService from '../src/sdk/services/oauthService';
+
+describe('sdk tests', () => {
+  let sdk: ExtensibilitySdk;
+
+  describe.only('receiving messages', () => {
+    var messageHandler: EventListenerOrEventListenerObject;
+    let originalEventListener: (event: string, handler: EventListenerOrEventListenerObject) => void;
+    let originalOpenPopup: (redirectUri?: string, state?: { [key: string]: string }) => void;
+
+    beforeEach(async () => {
+      window.setTimeout = jest.fn().mockImplementation(() => 123) as any;
+
+      originalOpenPopup = oauthService.openPopup;
+      oauthService.openPopup = jest.fn();
+
+      originalEventListener = window.addEventListener;
+      window.addEventListener = (event: string, handler: EventListenerOrEventListenerObject): void => {
+        if (event === 'message') {
+          messageHandler = handler;
+        }
+      };
+      runtime.application = {
+        api: {
+          client: { id: '' },
+          redirectUris: [],
+          scopes: [],
+        },
+      } as any;
+      runtime.origin = 'test.outreach.io';
+      sdk = new ExtensibilitySdk();
+      sdk.init();
+    });
+
+    afterEach(() => {
+      window.addEventListener = originalEventListener;
+      oauthService.openPopup = originalOpenPopup;
+      jest.restoreAllMocks();
+    });
+
+    it('will register message event listener', () => {
```

Comment:
Could we make the name a bit more descriptive?

Reply:
restored

Reply:
setting event listener is implementation detail. For which purpose is it set? The name should reflect that.

Reply:
purpose is to verify that event listener is registered

You know what - I will just remove this test

Reply:
removed